app.py

```diff
@@ -15,38 +72,55 @@
                        ).fillna('')
 
 
+# Generate answers
+for i in range(len(model_names)):
+    questions[model_names[i]] = questions.Question.apply(
+                                lambda x: generate_text(
+                                          x,
+                                          questions.Context[questions.Question == x].iat[0],
+                                          model_names[i],
+                                          model[i],
+                                          tokenizers[i],
+                                          minimum=len(questions.Answer[questions.Question == x].iat[0].split())+8))
+
+
 questions['Choice'] = ''
-cols = ['Answer'] + list('ABC')
-answers = {}
+answers = ['Answer'] + model_names
 
 
-for col in 'ABC':
-    questions[col] = questions.Answer.apply(lambda x: f'{col} {x}')
+# App
+with gr.Blocks() as test_gen:
+    with gr.Row():
 
+        with gr.Column(scale=1):
+            pass
 
-def get_answer(question, answer):
-    global questions
-    global answers
-    answers.update({question:int(answer == questions.Answer[questions.Question == question].iat[0])})
+        with gr.Column(scale=2):
 
+            gr.Markdown(f'## <p style="text-align: center;">IMITATION GAME</p>\n' +
+                        f'### <p style="text-align: center;">Choose answers NOT given by an AI model.')
 
-def set_score():
-    global answers
-    return str(sum(answers.values()))
+            for ind, question in enumerate(questions.Question.to_list(), start=1):
+                letters = list('ABCD')
+                options = list(set([questions[col][questions.Question == question].iat[0] for col in answers]))
+
+                gr.Markdown(f'### <p>{ind}. {question}</p>')
+
+                for letter, option in zip(letters, options):
+                    gr.Markdown(f'{letter}. {option}')
+
+                options = gr.State(SEPARATOR.join(options))
 
+                radio = gr.Radio(letters, type='index', show_label=False)
+                question = gr.State(question)
+                radio.change(fn=get_answer, inputs=[question, radio, options])
 
-with gr.Blocks() as demo:
-    for question in questions.Question.to_list():
-        radio = gr.Radio(list(set(
-                        [questions[col][questions.Question == question].iat[0] for col in cols])),
-                        label=question)
-        question = gr.State(question)
-        radio.change(fn=get_answer, inputs=[question, radio])
+            button = gr.Button(value='Get score')
+            score = gr.Markdown()
+            button.click(fn=set_score, outputs=score)
 
-    button = gr.Button(value='Get score')
-    score = gr.Markdown()
-    button.click(fn=set_score, outputs=score)
+        with gr.Column(scale=1):
+            pass
 
 
-if __name__ == "__main__":
```

Comment:
Would be a good idea to keep this line, so that we can import functions for testing without waiting for models and app to load